Sources/Fluent/Entity/Entity.swift

```diff
@@ -82,6 +84,10 @@ extension Entity {
     public static var name: String {
         return String(describing: self).lowercased()
     }
+    
+    public static var idType: Schema.Field.KeyType? {
```

Comment:
Hmm, `database` of `Entity` is an optional, which is not ideal in making this `Entity.idType` non-optional by relying on `database?.driver.idType`. It appears other cases of dealing with that in `Entity.swift` either return nil or throw an error, neither of which would really be ideal here. When trying to resolve `idType` I would say it is reasonable to treat it as a precondition that `database != nil` (with a message telling that you've probably forgotten to set the database of the entity)?

Further, there is no `idType` on Driver yet, only `idKey`. Did you mean I should introduce `idType` in there (I am in agreement, if yes)?